java/vector/src/main/java/org/apache/arrow/vector/NullVector.java

```diff
@@ -73,17 +74,19 @@ public NullVector(String name, int valueCount) {
    * @param name      name of the vector
    * @param fieldType type of Field materialized by this vector.
    */
+  @Deprecated
   public NullVector(String name, FieldType fieldType) {
-    this(new Field(name, fieldType, null));
+    this(name);
   }
 
   /**
    * Instantiate a NullVector.
    *
    * @param field field materialized by this vector.
    */
+  @Deprecated
   public NullVector(Field field) {
```

Comment:
I think to allow generic code it might be best to still allow the various field constructors, and just check that the field type is actually NULL, using https://github.com/apache/arrow/blob/0ce72675f4dd755b2696eb6597850b21df647bb8/java/memory/memory-core/src/main/java/org/apache/arrow/util/Preconditions.java#L130-L144

Reply:
Mostly so code that already has a field doesn't have to special case NullVector vs other vectors, and to keep the various vectors somewhat consistent.

e.g. IntVector has these constructors too (arguably, it should also be validating the type!)

https://github.com/apache/arrow/blob/0402e306a9d9f57ff22c87bf8689b8e7203483e5/java/vector/src/main/java/org/apache/arrow/vector/IntVector.java#L48-L73